src/main/java/com/jcraft/jsch/OpenSshCertificateAwareIdentityFile.java

```diff
@@ -0,0 +1,285 @@
+package com.jcraft.jsch;
+
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+
+import static com.jcraft.jsch.OpenSshCertificateUtil.extractComment;
+import static com.jcraft.jsch.OpenSshCertificateUtil.extractKeyData;
+import static com.jcraft.jsch.OpenSshCertificateUtil.extractKeyType;
+import static com.jcraft.jsch.OpenSshCertificateUtil.getRawKeyType;
+import static com.jcraft.jsch.OpenSshCertificateUtil.toDateString;
+import static com.jcraft.jsch.OpenSshCertificateUtil.trimToEmptyIfNull;
+import static com.jcraft.jsch.Util.fromBase64;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * An {@link Identity} implementation that supports OpenSSH certificates.
+ *
+ * <p>
+ * This class handles SSH identity files that contain OpenSSH certificates, which combine a public
+ * key with additional metadata and restrictions signed by a certificate authority. It supports all
+ * standard OpenSSH certificate types including RSA, DSA, ECDSA, and Ed25519.
+ * </p>
+ *
+ * <p>
+ * The class can load certificates from file pairs (private key file + certificate file) and
+ * provides all the functionality needed for SSH authentication using certificates.
+ * </p>
+ */
+class OpenSshCertificateAwareIdentityFile implements Identity {
+
+  public static final String SSH_RSA_CERT_V01_AT_OPENSSH_DOT_COM = "ssh-rsa-cert-v01@openssh.com";
```

Comment:
Let's make everything public in this class package private unless it is an `@Override`.